tests/unit/test_client.py

```diff
@@ -562,6 +562,29 @@ def test_get_bucket_with_object_hit(self):
         parms = dict(urlparse.parse_qsl(qs))
         self.assertEqual(parms["projection"], "noAcl")
 
+    def test_get_bucket_default_retry(self):
+        from google.cloud.storage.bucket import Bucket
+        from google.cloud.storage._http import Connection
+
+        PROJECT = "PROJECT"
+        CREDENTIALS = _make_credentials()
+        client = self._make_one(project=PROJECT, credentials=CREDENTIALS)
+
+        bucket_name = "bucket-name"
+        bucket_obj = Bucket(client, bucket_name)
+
+        with mock.patch.object(Connection, "api_request") as req:
+            client.get_bucket(bucket_obj)
+            req.assert_called_once_with(
```

Comment:
It is clearer to move the assertion back out of the `with` scope.

Also, ISTM we need to test that an explicitly-passed `retry` argument is honored.